core/src/components/datetime/utils/parse.ts

```diff
@@ -106,3 +108,20 @@ export const parseDate = (val: string | undefined | null): any | undefined => {
     tzOffset,
   };
 };
+
+export const clampDate = (date: string, min?: string, max?: string): string => {
+  const dateParts = parseDate(date);
```

Comment:
You could also optionally parse `date` in `datetime.tsx` so that this function takes in `DatetimeParts` across the board, but I don't have any strong opinions on this. The one benefit I see is consistency of arguments given that they are all dates.